sql/analyzer/stored_procedures.go

```diff
@@ -254,10 +251,105 @@ func applyProcedures(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, se
 		return n, transform.SameTree, nil
 	}
 
-	scope, err := loadStoredProcedures(ctx, a, n, scope, sel)
```

Comment:
Do we still need the `loadStoredProcedures` function? Looks like it's just called from one place after this change – the `assignRoutines` rule. I didn't dig very deep there, but figured you probably already investigated this.

Reply:
I decided to leave it there as `assignRoutines` still wants to load all of the stored procedures, so it's still useful to have it in a dedicated function.